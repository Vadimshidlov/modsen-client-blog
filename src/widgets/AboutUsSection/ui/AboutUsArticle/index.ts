export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
