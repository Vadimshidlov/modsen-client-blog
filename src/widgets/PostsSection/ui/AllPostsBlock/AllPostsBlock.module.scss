.allPosts__container {
  @media (max-width: 1380px) {
    width: 100%;
  }

  .allPostsHeader__container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 32px;

    @media (max-width: 768px) {
      margin-bottom: 16px;
    }

    .section__title {
      font-size: $fz36;
      font-weight: $bold;
      line-height: 48px;

      @media (max-width: 768px) {
        font-size: $fz28;
      }

      @media (max-width: 450px) {
        font-size: $fz22;
      }
    }

    .section__link {
      font-size: $fz16;
      font-weight: $regular;
      line-height: 28px;
    }
  }
}




