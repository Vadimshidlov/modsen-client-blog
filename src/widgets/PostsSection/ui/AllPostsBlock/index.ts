export { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock/AllPostsBlock';
