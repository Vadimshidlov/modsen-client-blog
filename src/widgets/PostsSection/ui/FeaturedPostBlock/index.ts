export { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock/FeaturedPostBlock';
