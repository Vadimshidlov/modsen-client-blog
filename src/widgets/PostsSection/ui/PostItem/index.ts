export { PostItem } from '@/widgets/PostsSection/ui/PostItem/PostItem';
