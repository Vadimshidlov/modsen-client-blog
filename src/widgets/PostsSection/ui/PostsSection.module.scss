.posts {
  padding-top: 128px;
  margin-bottom: 128px;
  display: flex;
  justify-content: space-between;
  column-gap: 28px;

  @media (max-width: 1380px) {
    flex-direction: column;
    align-items: center;
  }

  @media (max-width: 768px) {
    padding-top: 64px;
    margin-bottom: 64px;
  }

  @media (max-width: 450px) {
    padding-top: 32px;
    margin-bottom: 32px;
  }
}
