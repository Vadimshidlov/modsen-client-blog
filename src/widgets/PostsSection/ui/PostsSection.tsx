import React from 'react';
import { FeaturedPostBlock } from '@/widgets/PostsSection/ui/FeaturedPostBlock';
import { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock';
import styles from '@/widgets/PostsSection/ui/PostsSection.module.scss';
import container from '@/shared/styles/container.module.scss';

export function PostsSection() {
  return (
    <section className={`${styles.posts} ${container.container}`}>
      <FeaturedPostBlock />
      <AllPostsBlock />
    </section>
  );
}
