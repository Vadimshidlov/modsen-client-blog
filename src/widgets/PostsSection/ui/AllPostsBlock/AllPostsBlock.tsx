import React from 'react';
import Link from 'next/link';
import { useTranslations } from 'next-intl';
import { PostItem } from '@/widgets/PostsSection/ui/PostItem/PostItem';
import { POSTS } from '@/shared/constants/posts';
import styles from '@/widgets/PostsSection/ui/AllPostsBlock/AllPostsBlock.module.scss';

export function AllPostsBlock() {
  const t = useTranslations('AllPostsBlock');

  return (
    <section className={styles.allPosts__container}>
      <div className={styles.allPostsHeader__container}>
        <h3 className={styles.section__title}>{t('title')}</h3>
        <Link className={styles.section__link} href="/blog">
          <span>{t('linkButton')}</span>
        </Link>
      </div>
      {POSTS.slice(0, 4).map(({ id, postDate, author, title }) => (
        <Link key={id} className={styles.section__link} href={`/posts/${id}`}>
          <PostItem postDate={postDate} postText={title} authorName={author} />
        </Link>
      ))}
    </section>
  );
}
