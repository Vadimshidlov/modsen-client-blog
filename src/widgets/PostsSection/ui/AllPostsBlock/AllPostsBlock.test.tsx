import React from 'react';
import { render } from '@testing-library/react';
import { AllPostsBlock } from '@/widgets/PostsSection/ui/AllPostsBlock';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AllPostsBlock tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<AllPostsBlock />);
    expect(container).toMatchSnapshot();
  });
});
