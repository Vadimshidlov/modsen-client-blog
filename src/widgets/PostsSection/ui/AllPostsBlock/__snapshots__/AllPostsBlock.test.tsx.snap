// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPostsBlock tests should matches snapshot 1`] = `
<div>
  <section
    class="allPosts__container"
  >
    <div
      class="allPostsHeader__container"
    >
      <h3
        class="section__title"
      >
        Mocked title
      </h3>
      <a
        class="section__link"
        href="/blog"
      >
        <span>
          Mocked linkButton
        </span>
      </a>
    </div>
    <a
      class="section__link"
      href="/posts/1"
    >
      <div
        class="post__container"
      >
        <span
          class="post__authorInfo"
        >
          By 
          <span>
            Floyd Miles
          </span>
           
          May 23, 2024
        </span>
        <h4
          class="post__text"
        >
          Step-by-step guide to choosing great font pairs
        </h4>
      </div>
    </a>
    <a
      class="section__link"
      href="/posts/2"
    >
      <div
        class="post__container"
      >
        <span
          class="post__authorInfo"
        >
          By 
          <span>
            Dianne Russell
          </span>
           
          Aug 23, 2024
        </span>
        <h4
          class="post__text"
        >
          Design tips for designers that cover everything you need
        </h4>
      </div>
    </a>
    <a
      class="section__link"
      href="/posts/3"
    >
      <div
        class="post__container"
      >
        <span
          class="post__authorInfo"
        >
          By 
          <span>
            Jenny Wilson
          </span>
           
          Aug 23, 2024
        </span>
        <h4
          class="post__text"
        >
          How to build rapport with your web design clients
        </h4>
      </div>
    </a>
    <a
      class="section__link"
      href="/posts/4"
    >
      <div
        class="post__container"
      >
        <span
          class="post__authorInfo"
        >
          By 
          <span>
            Leslie Alexander
          </span>
           
          Aug 23, 2024
        </span>
        <h4
          class="post__text"
        >
          Logo design trends to avoid in 2022
        </h4>
      </div>
    </a>
  </section>
</div>
`;
